src/core/mac/mac_frame.hpp

```diff
@@ -1561,13 +1612,20 @@ class BeaconPayload
      */
     const otExtendedPanId &GetExtendedPanId(void) const { return mExtendedPanId; }
 
+    /**
+     * This method sets the Extended PAN ID field.
+     *
+     * @param[in]  aExtPanId  An Extended PAN ID.
+     *
+     */
+    void SetExtendedPanId(const otExtendedPanId &aExtPanId) { mExtendedPanId = aExtPanId; }
+
 private:
     uint8_t         mProtocolId;
     uint8_t         mFlags;
     char            mNetworkName[kMaxSize];
     otExtendedPanId mExtendedPanId;
 } OT_TOOL_PACKED_END;
-#endif // OPENTHREAD_CONFIG_MAC_BEACON_PAYLOAD_PARSING_ENABLE
```

Comment:
Suggest instead of defining and using `kMaxSize` in `BeaconPayload` we directly use the definition in `NetworkName` (makes it clear that it is max size of `NetworkName` and avoid re-defining same constant).

```c++
char mNetworkName[MeshCop::NetworkName::kMaxSize];
```